functions.j2

```diff
@@ -134,39 +134,47 @@ function {{ function }}() {
-  declare "SKIP_{{ '${' }}{{ tool | replace("-", "_") }}_upper}={{ '${' }}!{{ tool | replace("-", "_") }}_skip_env_var:-false}"
-
-  ## Methods to skip {{ tool | replace("-", "_") }}
-  if [[ "{{ '${' }}{{ tool | replace("-", "_") }}_skipped:-false}" == "argument" ]]; then
-    _feedback WARNING "Skipping {{ tool | replace("-", "_") }} due to {{ '${' }}{{ tool | replace("-", "_") }}_skip_argument}"
-    unset '{{ tool | replace("-", "_") }}_skipped'
+  {%- for security_tool in security_tools %}
+  {{ security_tool | replace("-", "_") }}_skip_argument="--skip-{{ security_tool | replace("-", "_") }}"
+  {{ security_tool | replace("-", "_") }}_upper=$(echo "{{ security_tool | replace("-", "_") }}" | tr '[:lower:]' '[:upper:]')
+  {{ security_tool | replace("-", "_") }}_skip_env_var="SKIP_{{ '${' }}{{ security_tool | replace("-", "_") }}_upper}"
+  declare "SKIP_{{ '${' }}{{ security_tool | replace("-", "_") }}_upper}={{ '${' }}!{{ security_tool | replace("-", "_") }}_skip_env_var:-false}"
+
+  ## Methods to skip {{ security_tool | replace("-", "_") }}
+  if [[ "{{ '${' }}{{ security_tool | replace("-", "_") }}_skipped:-false}" == "argument" ]]; then
+    _feedback WARNING "Skipping {{ security_tool | replace("-", "_") }} due to {{ '${' }}{{ security_tool | replace("-", "_") }}_skip_argument}"
+    unset '{{ security_tool | replace("-", "_") }}_skipped'
   # If the "skip" environment variable was set to true, skip the security scan
-  elif [[ "${!{{ tool | replace("-", "_") }}_skip_env_var,,}" == "true" ]]; then
-    _feedback WARNING "Skipping {{ tool | replace("-", "_") }} due to the {{ '${' }}{{ tool | replace("-", "_") }}_skip_env_var} environment variable value"
+  elif [[ "${!{{ security_tool | replace("-", "_") }}_skip_env_var,,}" == "true" ]]; then
+    _feedback WARNING "Skipping {{ security_tool | replace("-", "_") }} due to the {{ '${' }}{{ security_tool | replace("-", "_") }}_skip_env_var} environment variable value"
   # If the security scan was already run successfully in an non-interactive
   # shell, don't run it again
-  elif [[ -r "/tmp/{{ tool | replace("-", "_") }}_complete" ]]; then
-    _feedback INFORMATIONAL "Skipping {{ tool | replace("-", "_") }} because it was already run on $(date -d @"$(cat /tmp/{{ tool | replace("-", "_") }}_complete)")"
+  elif [[ -r "/tmp/{{ security_tool | replace("-", "_") }}_complete" ]]; then
+    _feedback INFORMATIONAL "Skipping {{ security_tool | replace("-", "_") }} because it was already run on $(date -d @"$(cat /tmp/{{ security_tool | replace("-", "_") }}_complete)")"
   # Otherwise, attempt to run the security tool
   else
-    if [[ -x $(which {{ tool }}) ]]; then
-      {{ security_tools[tool].command }} &>/tmp/{{ tool | replace("-", "_") }}_output
-      process_command_exit_status "${?}" "{{ tool | replace("-", "_") }}" "{{ security_tools[tool].description }}"
+    if [[ -x $(which {{ security_tool }}) ]]; then
+      security_tool_command='{{ security_tools[security_tool].command }}'
```

Comment:
This is the primary functionality change for this PR.  It loops over the provided customizations (from `easy_infra.yml`) and uses string concatenation to create a final `security_tool_command`, which is then (safely) `eval`'d.  This approach is focused on allowing runtime customization using environment variables.  An example is in `docs/Terraform/index.rst`.